src/io/perun.clj

```diff
@@ -105,74 +145,103 @@
-                     add-filedata
-                     (trace :io.perun/images-resize))
-          updated-files (pod/with-call-in @pod
-                         (io.perun.contrib.images-resize/images-resize ~(.getPath tmp) ~files ~options))]
-      (perun/report-debug "images-resize" "new resized images" updated-files)
+          metas (trace :io.perun/images-resize
+                       (meta-by-ext fileset [".png" ".jpeg" ".jpg"]))
+          updated-metas (pod/with-call-in @pod
+                         (io.perun.contrib.images-resize/images-resize ~(.getPath tmp) ~metas ~options))]
+      (perun/report-debug "images-resize" "new resized images" updated-metas)
       (-> fileset
           (commit tmp)
-          (pm/set-meta updated-files)))))
-
-(defn meta-by-ext
-  [fileset file-exts]
-  (->> fileset
-       boot/ls
-       (boot/by-ext file-exts)
-       (keep pm/meta-from-file)))
+          (pm/set-meta updated-metas)))))
+
+;; These are modified diff functions from boot. They can be removed if/when
+;; this PR is merged: https://github.com/boot-clj/boot/pull/566
+(defn- diff-tree
+  [tree props]
+  (let [->map #(select-keys % props)]
+    (reduce-kv #(assoc %1 %2 (->map %3)) {} tree)))
+
+(defn- diff*
+  [{t1 :tree :as before} {t2 :tree :as after} props]
+  (if-not before
+    {:added   after
+     :removed (assoc after :tree {})
+     :changed (assoc after :tree {})}
+    (let [props        (or (seq props) [:id])
+          d1           (diff-tree t1 props)
+          d2           (diff-tree t2 props)
+          [x y z]      (map (comp set keys) (data/diff d1 d2))
+          changed-keys (set/union (set/intersection x y)
+                                  (set/intersection (set/union x y) z))]
+      {:added   (->> (set/difference y x changed-keys) (select-keys t2) (assoc after :tree))
+       :removed (->> (set/difference x y changed-keys) (select-keys t1) (assoc after :tree))
+       :changed (->> changed-keys                      (select-keys t2) (assoc after :tree))})))
+
+(defn- diff
+  [before after & props]
+  (let [{:keys [added changed]}
+        (diff* before after props)]
+    (update-in added [:tree] merge (:tree changed))))
 
 (defn content-pre-wrap
   "Wrapper for input parsing tasks. Calls `parse-form` on new or changed
-  files with extensions in `file-exts`, adds `tracer` to `:io.perun/trace`
-  and writes html files for subsequent tasks to process, if desired. Pass
+  files with extensions in `extensions`, adds `tracer` to `:io.perun/trace`
+  and writes files for subsequent tasks to process, if desired.
+  `output-extension` can be used to indicate the type of file the task produces,
+  or pass `nil` to overwrite with a file of the same name and extension. Pass
   `pod` if one is needed for parsing"
-  [parse-form file-exts tracer & [pod]]
-  (let [tmp     (boot/tmp-dir!)
-        prev-fs (atom nil)]
+  [parse-form-fn extensions output-extension tracer options & [pod]]
```

Comment:
I'll go with a map - I don't much like the `(a-fn :key1 "foo" :key2 "bar")` style.  Feel free to voice any objections to that, though.